### ISSUE TITLE
Add support JS objects for `extends` config option

### ISSUE THREAD

**Issue by ybiquitous** [stylelint/stylelint#6952]
### What is the problem you're trying to solve?

First, assume that there is a pure ESM package for a sharable config named `foo-config`:

`foo-config/package.json`:

```json
{
  "name": "foo-config",
  "version": "1.0.0",
  "type": "module",
  "main": "index.js"
}
```

`foo-config/index.js`:

```js
export default {
	rules: {
		'color-no-invalid-hex': true,
	},
};
```

Next, consider a case using `foo-config` via npm:

`package.json`:

```json
{
  "dependencies": {
    "foo-config": "^1.0.0",
    "stylelint": "^15.8.0"
  }
}
```

`.stylelintrc.mjs`:

```js
import foo from 'foo-config';

export default {
	// Not working => Error [ERR_REQUIRE_ESM]: require() of ES Module ...
	extends: 'foo-config',

	// Still not working => TypeError: Expected `moduleId` to be of type `string`, got `object`
	extends: foo,
};
```

Running `stylelint` like this raises the errors above:

```shell
echo 'a { color: #f4 }' | npx stylelint --stdin
```

This means that Stylelint cannot use a pure ESM sharable config with the `extends` option.

Related: #5291

### What solution would you like to see?

I expect Stylelint to work even when using JS objects with `extends` instead of strings (e.g., package name or path). For example:

```console
$ echo 'a { color: #f4 }' | npx stylelint --stdin

<input css qXAh0i>
 1:12  ✖  Unexpected invalid hex color "#f4"  color-no-invalid-hex

1 problem (1 error, 0 warnings)
```

I find it works by applying the following patch:

```diff
--- a/lib/augmentConfig.js
+++ b/lib/augmentConfig.js
@@ -178,7 +178,13 @@ async function extendConfig(stylelint, config, configDir, rootConfigDir, filePat
 	let resultConfig = originalWithoutExtends;
 
 	for (const extendLookup of normalizedExtends) {
-		const extendResult = await loadExtendedConfig(stylelint, configDir, extendLookup);
+		let extendResult;
+
+		if (typeof extendLookup === 'string') {
+			extendResult = await loadExtendedConfig(stylelint, configDir, extendLookup);
+		} else if (typeof extendLookup === 'object' && extendLookup !== null) {
+			extendResult = { config: extendLookup };
+		}
 
 		if (extendResult) {
 			let extendResultConfig = extendResult.config;
```

This patch simply uses JS objects if they are passed.

What do you think about this idea? Is there some pitfall?

---

For your information, I refer to the current spec of `extends`:

https://github.com/stylelint/stylelint/blob/16754b1dab7590238f9b842ebd25678f8204ae9d/docs/user-guide/configure.md?plain=1#L215-L219
